docs/configuration.rst

```diff
@@ -4,59 +4,197 @@
+                  be an arbitrary string or one of the magic
+                  keywords ``AUTO`` (default), ``GIT``, or
+                  ``HG``. By default, ``shub`` will
+                  auto-detect your version control system and
+                  use its branch/commit ID as version.
+``apikey``        API key to use for deployments. You will      global only
+                  typically not have to touch this setting as
+                  it will be configured inside
+                  ``~/.scrapinghub.yml`` in your home
+                  directory, via ``shub login``.
+================  ============================================  ===============
+
+
+.. _`Scrapy Cloud stack`: https://helpdesk.scrapinghub.com/support/solutions/articles/22000200402-scrapy-cloud-stacks
+
 
-    # Use git branch/commit as version when deploying. Other possible values
-    # are AUTO (default) or HG
-    version: GIT
+Example configurations
+----------------------
 
-``projects`` is a mapping from human-friendly names to project IDs. This allows
-you to run ``shub deploy prod`` instead of ``shub deploy 33333``. ``version``
-is a string to be used as project version information when deploying to Scrapy
-Cloud. By default, ``shub`` will auto-detect your version control system and
-use its branch/commit ID as version.
+Custom requirements file and fixed version information::
 
-Typically, you will configure only your API keys and the version specifier in
-the global ``~/.scrapinghub.yml``, and keep project configuration in the local
-``project_dir/scrapinghub.yml``. However, sometimes it can be convenient to
-also configure projects in the global configuration file. This allows you to
-use the human-friendly project identifiers outside of the project directory.
+    project: 12345
+    requirements:
+      file: requirements_scrapinghub.txt
+    version: 0.9.9
+
+Custom Scrapy Cloud stack, requirements file and additional private
+dependencies::
+
+    project: 12345
+    stack: scrapy:1.1
+    requirements:
+      file: requirements.txt
+      eggs:
+        - privatelib.egg
+        - path/to/otherlib.egg
+
+Using the latest Scrapy 1.3 stack in staging and development, but pinning the
+production stack to a specific release::
+
+    projects:
+      default: 12345
+      staging: 33333
+      prod:
+        id: 44444
+        stack: scrapy:1.3-py3-20170322
+ 
+    stack: scrapy:1.3-py3
+
+Using a custom Docker image::
+
+    projects:
+      default: 12345
+      prod: 33333
+
+    image: true
+
+Using a custom Docker image only for the development project::
+
+    projects:
+      default:
+        id: 12345
+        image: true
+      prod: 33333
+
+Using a custom Docker image in staging and development, but a Scrapy Cloud
+stack in production::
+
+    projects:
+      default: 12345
+      staging: 33333
+      prod:
+        id: 44444
+        image: false
+        stack: scrapy:1.3-py3-20170322
+
+    image: true
```

Comment:
It was probably discussed earlier, but what if there's a global `image` and global `stack`? I'm pretty sure `image` has higher priority, but it can be non-obvious for clients.